### ISSUE TITLE
Incoming nightly breakage

### ISSUE THREAD

**Issue by oli-obk** [mit-pdos/noria#189]
While your crate does nothing wrong per se, the incoming breakage may be somewhat annoying, and you can get ahead of the breakage.

The PR that will break this crate: https://github.com/rust-lang/rust/pull/94081

The breakage: https://crater-reports.s3.amazonaws.com/pr-94081/try%23e251c42c969d6f8dbaae00e1028489c3a29f8789/reg/noria-common-0.6.0/log.txt

The fix is to move https://github.com/mit-pdos/noria/blob/465184ee4b57a2791d41b0f12fb8f71c45bae6b6/noria/src/controller.rs#L358 and https://github.com/mit-pdos/noria/blob/465184ee4b57a2791d41b0f12fb8f71c45bae6b6/noria/src/controller.rs#L179 into their own module, so that the type alias doesn't try to get inferred at the same time as the `impl Trait` usage in the return type.

On a positive note: https://github.com/rust-lang/rust/issues/65442 is now very much on my radar, with the PR that breaks your code this becomes actually fixable without a complex hack!

**Comment by jonhoo** [comment]
Thanks for the heads up! This repository is basically archived at this point, and the upstream crate hasn't been updated in a while, so I probably won't go through and fix this, but this should be helpful to anyone who does want to pick up the code. So I'll leave it open :)

**Comment by muety** [comment]
So the project is not under active development anymore?

**Comment by jonhoo** [comment]
No, I haven't worked on this since I graduated back in November 2020, and I don't *think* anyone's currently continuing the research in my lab at MIT either. This repository was, and still is, a research prototype :)

**Comment by ogbanugot** [comment]
Hi @jonhoo based on the breaking changes coming in can you help state clearly which version of nightly one needs to run this project smoothly. Thanks a lot.

**Comment by jonhoo** [comment]
Ah, that's a good question. I last worked on this end of November 2020, so a nightly from around there is likely to work. I don't have an exact date though.
